src/Caching/CachedMap.purs

```diff
@@ -145,3 +148,16 @@ size (CachedMap { ref }) = Map.size <$> Ref.read ref
 
 instantAddMs ∷ Instant -> Milliseconds -> Instant
 instantAddMs x ms = unsafePartial $ fromJust $ instant (unInstant x <> ms)
+
+-- | Check if CachedMap has been initialized
+-- |
+-- | Returns True only when underlying Map is not empty, and all avars are filled
+isInitialized ∷ ∀  m k a. MonadEffect m => CachedMap k a -> m Boolean
+isInitialized (CachedMap { ref }) = liftEffect do
+  cache <- Ref.read ref
+  let mapNotEmpty = (not <<< Map.isEmpty) cache
+      allAvars = _.value <$> Map.values cache
+      allAvarsFilled = do
+        results <- traverse AVar.tryRead allAvars
+        pure $ all isJust results
+  (&&) <$> pure mapNotEmpty <*> allAvarsFilled
```

Comment:
I don't think this function is a good idea. `CachedMap` is supposed to be used for caching values that you don't know you will need until they are requested for the first time. If you know beforehand which values you will need you probably want to use multiple `SynchedValue`s (or one `SynchedValue` holding all the values).

Even if we considered the case where you really have a space of keys that you're interested in, but require only some of them from startup, the behavior of this function would be quite unstable as it would rely on the ordering of the client reads.

Reply:
That makes sense, I'll drop it